state/cleanup.go

```diff
@@ -177,12 +177,9 @@ func (st *State) cleanupModelsForDyingController() (err error) {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	for _, env := range models {
-
-		if env.Life() == Alive {
-			if err := env.Destroy(); err != nil {
-				return errors.Trace(err)
-			}
+	for _, model := range models {
```

Comment:
Looking at DestroyOps() won't this cause an early failure with "errModelNotAlive" in the case that we run this and one of the models is already dying but not dead yet? eg:
juju add-model foo
juju add-model bar
juju destroy-model foo
juju destroy-controller